src/EventStore.Core/Certificates/CertificateUtils.cs

```diff
@@ -96,18 +97,36 @@ private static bool TryReadPkcs12CertificateBundle
 				}
 
 				// if the above attempt failed, try to load the certificate via the standard constructor
-				var certificate = new X509Certificate2(certificatePath, password);
+				var certificate = new X509Certificate2(certificatePath, certificatePassword);
 				if (!certificate.HasPrivateKey)
 					throw new NoCertificatePrivateKeyException();
 				return (certificate, null);
 			}
 
-			var privateKey = Convert.FromBase64String(string.Join(string.Empty, File.ReadAllLines(privateKeyPath)
-				.Skip(1)
-				.SkipLast(1)));
+			string[] allLines = File.ReadAllLines(privateKeyPath);
+			var header = allLines[0].Replace("-", "");
+			var privateKey = Convert.FromBase64String(string.Join(string.Empty, allLines.Skip(1).SkipLast(1)));
 
 			using var rsa = RSA.Create();
-			rsa.ImportRSAPrivateKey(new ReadOnlySpan<byte>(privateKey), out _);
+			switch (header) {
+				case "BEGIN PRIVATE KEY":
+					rsa.ImportPkcs8PrivateKey(new ReadOnlySpan<byte>(privateKey), out _);
+					break;
+				case "BEGIN RSA PRIVATE KEY":
+					rsa.ImportRSAPrivateKey(new ReadOnlySpan<byte>(privateKey), out _);
+					break;
+				case "BEGIN ENCRYPTED PRIVATE KEY":
+					if (privateKeyPassword is null) {
+						throw new ArgumentException(
+							"Need encryption password to read encrypted node private key");
+					}
+					rsa.ImportEncryptedPkcs8PrivateKey(new ReadOnlySpan<byte>(Encoding.UTF8.GetBytes(privateKeyPassword)), new ReadOnlySpan<byte>(privateKey), out _);
+					break;
+				default:
+					throw new InvalidFileException("Node private key file format is unrecognizable");
+					
```

Comment:
suggestion: after loading the key, check if it is an RSA key and throw a `NotSupportedException` if it's not.

Some time back, i had tested elliptical curve keys and it was not stable causing native code crashes. i don't know if the state has changed, but for the scope of this PR, we can just prevent use of non-RSA keys.

cc: @hunter1703 @alexeyzimarev @hayley-jean

Reply:
EDIT: ignore the above. just noticed we use rsa.Import*, so if we're loading an EC key, it'll throw an exception